inventory/main/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java

```diff
@@ -47,25 +45,40 @@ public Optional<MonitoringSystemDTO> findBySystemId(String systemId, String tena
         return systemRepository.findBySystemIdAndTenantId(systemId, tenantId).map(mapper::modelToDTO);
     }
 
-    public void addMonitoringSystemFromHeartbeat(TenantLocationSpecificHeartbeatMessage message) throws LocationNotFoundException {
+    public Optional<MonitoringSystemDTO> findById(long systemId, String tenantId) {
+        return systemRepository.findByIdAndTenantId(systemId, tenantId).map(mapper::modelToDTO);
+    }
+
+    public Optional<MonitoringSystemDTO> findBySystemId(long systemId) {
+        return systemRepository.findById(systemId).map(mapper::modelToDTO);
+    }
```

Comment:
Looks like the parameter should be called `id`.

```suggestion
    public Optional<MonitoringSystemDTO> findBySystemId(long id) {
        return systemRepository.findById(id).map(mapper::modelToDTO);
    }
```